sip/sip-005-blocks-and-transactions.md

```diff
@@ -0,0 +1,1164 @@
+evaluate than had they been defined as Clarity code.
+
+#### Post-Condition Limitations
+
+Post-conditions are not free, since validating, evaluating, and storing them
+carries non-zero computational cost that scales linearly in the number of
+post-conditions.  In addition, as mentioned, post-conditions only apply
+to the originating account's assets.  If the user wants post-conditions
+on other accounts or on non-asset state, they would need to be expressed in the Clarity code body.
+Post-conditions are _not_ meant to be (or become) a domain-specific language for
+encoding arbitrary post-conditions on arbitrary accounts -- they are _solely_
+meant to help preserve the safety of the _originating account_'s assets.
+
+Treating post-conditions as first-class transaction component strikes a balance
+between the need for complex abort logic and the need for cheap but effective smart contract 
+bug mitigation.  Any transaction's Clarity code body can already implement complex
+sanity checks to trigger an abort if certain conditions are met.  But, we expect
+that most users will send type-2 transactions to move assets to/from their
+(originating) accounts and perform atomic swaps, so most such sanity checks would only apply to their
+accounts in the first place.  By making this common case as easy as possible to
+encode in transactions and as cheap as possible to evaluate (cheaper than
+expressing directly in Clarity), the protocol encourages people to use it to protect
+even low-value accounts.
+
+### Transaction Fees
+
+Validating and processing a transaction carries non-zero cost, which must be
+paid for by a materialized account listed in the transaction.  In most
+blockchains, this fee is paid for in the blockchain's native token.  The purpose
+of the token in these cases is to act as a _decentralized rate-limiter_ -- as
+more and more transactions are issued, users must pay higher and higher fees to
+bid on their acceptance into a block.
+
+Unlike most blockchains, the Stacks blockchain offers the following separate payment
+methods for transaction fees:
+
+* **Originator pays in STX**.  This is the most straightforward way of paying
+  for a transaction.  The originating account that authorizes the transaction is
+an existing standard account, and that standard account holds a quantity of STX
+tokens that will be debited when the transaction's state-transitions are
+materialized.  This is how most blockchains work today.
+
+* **Originator pays in an on-chain asset**.  Unlike most blockchains, the Stacks
+  blockchain has first-class support for users to pay transaction fees via _any_
+on-chain asset.  In this payment mode, the originating account that authorizes
+the transaction is an existing standard account (as before), but instead of
+being debited STX, the transaction offers miners another asset instead, as well
+as a proposed _exchange rate_ between that asset and STX.  The offerred asset and the
```

Comment:
As a miner evaluating a transaction paid with an on-chain asset, my decision will be entirely relying on this **exchange rate**. Where is this value coming from? Do I have any insurance that I'll be able to liquidate this asset at this rate, knowing that a few minutes are separating `t(tx-issued)` and `t(tx-mined)`?